src/Sulu/Component/Persistence/EventSubscriber/ORM/TimestampableSubscriber.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Sulu\Component\Persistence\EventSubscriber\ORM;
+
+use Doctrine\ORM\Events;
+use Doctrine\ORM\EntityManager;
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Event\LifecycleEventArgs;
+use Doctrine\Common\Persistence\Event\LoadClassMetadataEventArgs;
+use Sulu\Component\Persistence\Model\TimestampableInterface;
+
+/**
+ * Manage the timestamp fields on models implementing the
+ * TimestampableInterface.
+ */
+class TimestampableSubscriber implements EventSubscriber
+{
+    const CREATED_FIELD = 'created';
+    const CHANGED_FIELD = 'changed';
+
+    public function getSubscribedEvents()
+    {
+        return array(
+            Events::loadClassMetadata,
+            Events::preUpdate,
+            Events::prePersist,
+        );
+    }
+
+    public function loadClassMetadata(LoadClassMetadataEventArgs $event)
+    {
+        $metadata = $event->getClassMetadata();
+        $reflection = $metadata->getReflectionClass();
+
+        if ($reflection->implementsInterface('Sulu\Component\Persistence\Model\TimestampableInterface')) {
+            if (!$metadata->hasField(self::CREATED_FIELD)) {
```

Comment:
Should chaeck to ensure that the object has the *property* before mapping to it (or will the ORM do this?)